src/cmd-buildprep

```diff
@@ -151,6 +151,10 @@ class S3Fetcher(Fetcher):
     def fetch_impl(self, url, dest):
         assert url.startswith("s3://")
         bucket, key = url[len("s3://"):].split('/', 1)
+        # if the key contains a dir make sure it exists so that download_file could write temp files
+        if "/" in key:
+            maybeMissingDirs = key.split("/")[:-1]
+            os.makedirs(os.path.join(maybeMissingDirs), exist_ok=True)
```

Comment:
Let's just move this to `fetch()` as the one-liner:

```python
os.makedirs(os.path.dirname(dest), exist_ok=True)
```

?

Reply:
Good idea, implemented in 383e0266054e9e68661a3bd5b754cd8a270deebc